.githooks/pre-commit

```diff
@@ -1,25 +1,45 @@
 #!/usr/bin/env bash
-# Call the source-format script
 exec 1>&2
 if [[ -v GIT_HOOKS_OFF ]]; then
 	echo "**Ignoring precommit / git-hooks**"
 	exit 0
 fi
 
-cd scripts
-if ! ./source-format.sh; then
+hash pre-commit 2>/dev/null || {
 	cat <<\EOF
 ==============================================================================
- Error: invalid source-format
+Error: the 'pre-commit' framework https://pre-commit.com/ is not available
 ==============================================================================
 
-* You can inspect the current format-issues here: 'scripts/source-format.log'
+You can install it via the Python Package Index:
 
-* Run the source-formatter / fix the source-format issues
+  pip3 install --user pre-commit
```

Comment:
Additional step for commit message check:
```
pre-commit install --hook-type commit-msg
```

Reply:
Doesn't ``pre-commit`` auto-install these on first-run? Along with all the others?

Reply:
Could be it was only explicitly needed, as I was adding the commit-msg hook after the regular install.

But if we have our own pre-commit script, we need a hook in Git to trigger on commit-msg.